### PR TITLE
fix: install current package as local dependency

### PR DESCRIPTION
**Pull request by ThibaultFy**
Signed-off-by: ThibaultFy <50656860+ThibaultFy@users.noreply.github.com>

## Related issue

`#` followed by the number of the issue
https://app.asana.com/0/1190766234519625/1202828693058876/f

## Please check if the PR fulfills these requirements

- [ ] If the feature has an impact on the user experience, the [changelog](https://github.com/substra/substrafl/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification

### CODE REVIEW

File: substrafl/remote/register/register.py [hunk 1]
```diff
@@ -85,7 +85,9 @@ def _copy_local_packages(
     for dependency_path in local_dependencies:
         dest_path = path / dependency_path.name
         if dependency_path.is_dir():
-            shutil.copytree(dependency_path, dest_path)
+            shutil.copytree(
+                dependency_path, dest_path, ignore=shutil.ignore_patterns("local-worker*", "tmp_substrafl*")
```
Comment (oleobal) [inline substrafl/remote/register/register.py:89]: Ignoring local-worker is a substrafl solution for a substra problem, maybe it should be solved there

Comment (ThibaultFy) [reply substrafl/remote/register/register.py]: Done in https://github.com/Substra/substra/pull/314 :)

File: substrafl/remote/register/register.py [hunk 2]
```diff
@@ -285,7 +287,7 @@ def register_algo(
     Returns:
         str: Substra algorithm key.
     """
-    with tempfile.TemporaryDirectory(dir=str(Path.cwd().resolve()), prefix="substrafl_") as operation_dir:
+    with tempfile.TemporaryDirectory(dir=str(Path.cwd().resolve()), prefix="tmp_substrafl_") as operation_dir:
```
[the remark this answers is not in the document]
Comment (ThibaultFy) [reply substrafl/remote/register/register.py]: Done, thanks !

File: substrafl/dependency.py [hunk 3]
```diff
@@ -12,6 +12,10 @@
 class Dependency(BaseModel):
     """Dependency pydantic class.
 
+
+    .. note:: If you are using your current package as a local dependencies, be aware that folders named
+        **local-worker** or with **tmp_substrafl** as prefix are ignored during the installation.
```
Comment (oleobal) [inline substrafl/dependency.py:17]: Is it possible to template this string with `substra.sdk.backends.local.LOCAL_WORKER_DIR` and `TMP_SUBSTRAFL_PREFIX`? 🤔

**Review by oleobal — APPROVED** [review]

**Pull request merged**